architecture/design/multi-region-2DC-deployment.md

```diff
@@ -216,13 +216,61 @@ Upon receiving a `ReplicateMsg` that is a part of a multi-row transaction, the *
+
+When a consumer polls for changes, it sends to the producer the latest replicated op id, and this information is saved on the producer's `cdc_state` table. The producer will take this information and call `Log::SetMinimumOpIdToRetain()`. This will in turn set the private variable `minimum_op_idx_to_retain_` which will be used by the Log garbage collector only when enable_log_retention_by_op_idx is set to true.
+
+`Log::SetMinimumOpIdToRetain()` will be called every time a cdc stream’s last replicated op id is updated (in `CDCServiceImpl::UpdateCheckpoint`) and every time the cdc service starts.
+
+## Setting up the op id retention policy
+For this policy to be active, all the tservers in the producing universe will have to be started with `--enable_log_retention_by_op_idx=true`. Initially, if no cdc stream is setup for a specific table, the policy will remain disabled for all the tablets in the table. Once the first cdc stream is setup, `Log::SetMinimumOpIdToRetain()` will be called with a value of 0. This will effectively disable garbage collection of log files until the consumer updates the latest replicated op id.
+
+Since we don’t have a mechanism to bootstrap a consumer once necessary log files have been deleted, this design only works when a 2DC consumer is setup soon after a table has been created in the producer. By default, `log_min_seconds_to_retain` is `300`, but we will allow users to modify this value for a specific table through the `yb-admin` command. That way, only log files for a specific table are retained for a long time so that when a cdc stream is created, all the records are available in the log files.
+
+## Executing the minimum op id retention policy
+We currently decide which log files to delete in Log::GetSegmentsToGCUnlocked(int64_t min_op_idx, SegmentSequence* segments_to_gc). This function gets called by Log::GC(int64_t min_op_idx, int32_t* num_gced).
+
+The list of log segments that are candidates to be GCed is generated by `LogReader::GetSegmentPrefixNotIncluding`. This method takes an op id index as a parameterto generate a list of log segments with log entries earlier than the specified op id. Since we will have a couple of new flags `log_max_seconds_to_retain` and `log_stop_retaining_min_disk_mb` that are used to override the op id retention policy, we need to be careful so that we don’t delete a log segment that contains unflushed entries since this will violate our durability guarantee. `LogReader::GetSegmentPrefixNotIncluding` will have to be aware of the two different indices: `min_flushed_index` and `min_cdc_replicated_index`. If all the entries in a log file are greater than `min_replicated_index`, and less than `min_flushed_index`, then this file could be GCed if its age is older than `log_max_seconds_to_retain` or if the storage available for logs is less than `log_stop_retaining_min_disk_mb`.
+## Updating the last replicated op id
```

Comment:
We can remove this section since you've already written about updating the op id during polling in the `Last replicated op id` section.